test/functional/feature_assumeutxo.py

```diff
@@ -202,7 +202,15 @@ def test_snapshot_with_less_work(self, dump_output_path):
         assert_equal(node.getblockcount(), FINAL_HEIGHT)
         with node.assert_debug_log(expected_msgs=["[snapshot] activation failed - work does not exceed active chainstate"]):
             assert_raises_rpc_error(-32603, "Unable to load UTXO snapshot", node.loadtxoutset, dump_output_path)
-        self.restart_node(0, extra_args=self.extra_args[0])
+
+    def test_snapshot_block_invalidated(self, dump_output_path):
+        self.log.info("Test snapshot is not loaded when base block is invalid.")
+        node = self.nodes[0]
+        block_hash = node.getblockhash(SNAPSHOT_BASE_HEIGHT - 1)
+        node.invalidateblock(block_hash)
```

Comment:
> If the base block of the snapshot is marked invalid or part of an invalid chain

This existing test seems to focus on the base block being part of an invalid chain.  Is it also worth adding an explicit test for an invalid base block itself (or would it be overkill)?

Reply:
I think it's a bit overkill but it was easy to add and also shouldn't make the test considerably slower, so I added coverage for this as well.